### PR TITLE
enum-based SELECT_CASE_var

### PR DESCRIPTION
**Pull request by dareumnam**
Pull request overview
---------------------
 - Replaced **enum-based** SELECT_CASE_var if-then ladders with switch/case statements.
 - **string-based/int-based** SELECT_CASE_var if-then ladders will be replaced in a separate PR.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [X] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/SolarCollectors.cc [hunk 1]
```diff
@@ -1148,24 +1149,25 @@ namespace SolarCollectors {
...
-                                    ((inletTemp + outletTemp) * 0.5 - state.dataSurface->SurfOutDryBulbTemp(SurfNum));
-                    TestTypeMod = 1.0 / (1.0 - FRULpTest / (2.0 * mCpATest));
-
-                } else if (SELECT_CASE_var == TestTypeEnum::OUTLET) {
-                    FRULpTest = state.dataSolarCollectors->Parameters(ParamNum).eff1 +
-                                state.dataSolarCollectors->Parameters(ParamNum).eff2 * (outletTemp - state.dataSurface->SurfOutDryBulbTemp(SurfNum));
-                    TestTypeMod = 1.0 / (1.0 - FRULpTest / mCpATest);
-                }
+            switch (state.dataSolarCollectors->Parameters(ParamNum).TestType) {
+            case TestTypeEnum::INLET: {
```
Comment (mitchute) [inline src/EnergyPlus/SolarCollectors.cc:1153]: Question for my own edification: what is the "right" place for the `break` statement? I've always put them inside of the curly braces but it probably doesn't really matter. Bueller... Bueller...

Comment (amirroth) [reply src/EnergyPlus/SolarCollectors.cc:1153]: The braces are technically not even necessary, although they are good hygiene, so it doesn’t matter.

Comment (amirroth) [inline src/EnergyPlus/SolarCollectors.cc:1153]: Remove Enum from this enumeration name. Type and Enum are redundant.

File: src/EnergyPlus/SteamBaseboardRadiator.cc [hunk 2]
```diff
@@ -196,37 +196,35 @@ namespace SteamBaseboardRadiator {
...
-                    } else {
-                        ShowSevereError(state,
-                                        "SimSteamBaseboard: Errors in Baseboard=" +
-                                            state.dataSteamBaseboardRadiator->SteamBaseboard(BaseboardNum).EquipID);
-                        ShowContinueError(state,
-                                          format("Invalid or unimplemented equipment type={}",
-                                                 state.dataSteamBaseboardRadiator->SteamBaseboard(BaseboardNum).EquipType));
-                        ShowFatalError(state, "Preceding condition causes termination.");
-                    }
+                switch (state.dataSteamBaseboardRadiator->SteamBaseboard(BaseboardNum).EquipType) {
```
Comment (amirroth) [inline src/EnergyPlus/SteamBaseboardRadiator.cc:199]: In the case of a single if/else, if/else is faster than switch/case. Still remove SELECT_CASE_var wrapper.

File: src/EnergyPlus/SolarCollectors.cc (hunk 1 above)
Comment (amirroth) [reply src/EnergyPlus/SolarCollectors.cc:1153]: Also why are the type values in this enum SCREAMING?

File: src/EnergyPlus/ThermalComfort.cc [hunk 3]
```diff
@@ -2457,15 +2459,18 @@ namespace ThermalComfort {
...
-            {
-                auto const SELECT_CASE_var(state.dataGlobal->KindOfSim);
-                if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::DesignDay) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstSimpleComfort, "Aggregated over the Design Days");
-                } else if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::RunPeriodDesign) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstSimpleComfort, "Aggregated over the RunPeriods for Design");
-                } else if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::RunPeriodWeather) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstSimpleComfort, "Aggregated over the RunPeriods for Weather");
-                }
+            switch (state.dataGlobal->KindOfSim) {
```
Comment (amirroth) [inline src/EnergyPlus/ThermalComfort.cc:2462]: This could probably be replaced with a local `constexpr std::array<std::string_view>`

File: src/EnergyPlus/ThermalComfort.cc [hunk 4]
```diff
@@ -2655,15 +2660,18 @@ namespace ThermalComfort {
...
-            {
-                auto const SELECT_CASE_var(state.dataGlobal->KindOfSim);
-                if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::DesignDay) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstUnmetLoads, "Aggregated over the Design Days");
-                } else if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::RunPeriodDesign) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstUnmetLoads, "Aggregated over the RunPeriods for Design");
-                } else if (SELECT_CASE_var == DataGlobalConstants::KindOfSim::RunPeriodWeather) {
-                    addFootNoteSubTable(state, state.dataOutRptPredefined->pdstUnmetLoads, "Aggregated over the RunPeriods for Weather");
-                }
+            switch (state.dataGlobal->KindOfSim) {
```
Comment (amirroth) [inline src/EnergyPlus/ThermalComfort.cc:2663]: Same comment.

File: src/EnergyPlus/UnitarySystem.cc [hunk 5]
```diff
@@ -255,25 +255,28 @@ namespace UnitarySystems {
...
-                    }
-                } else if (SELECT_CASE_var == UnitarySysCtrlType::Load || SELECT_CASE_var == UnitarySysCtrlType::CCMASHRAE) {
-                    if (ZoneEquipment) {
-                        this->controlUnitarySystemtoLoad(
-                            state, 0, FirstHVACIteration, CompressorOn, OAUCoilOutTemp, HXUnitOn, sysOutputProvided, latOutputProvided);
-                    } else {
-                        this->controlUnitarySystemtoLoad(
-                            state, AirLoopNum, FirstHVACIteration, CompressorOn, OAUCoilOutTemp, HXUnitOn, sysOutputProvided, latOutputProvided);
-                    }
+            switch (this->m_ControlType) {
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:258]: Is there really a difference between the three cases here? I can't see one.

File: src/EnergyPlus/UnitarySystem.cc [hunk 6]
```diff
@@ -8304,42 +8311,42 @@ namespace UnitarySystems {
...
-                        return;
-                    if (!state.dataUnitarySystems->HeatingLoad && !state.dataUnitarySystems->CoolingLoad &&
-                        (state.dataUnitarySystems->MoistureLoad >= 0.0 || state.dataUnitarySystems->MoistureLoad > LatOutputOff))
-                        return;
-                } else {
-                    // should never get here
-                }
+            switch (state.dataHeatBalFanSys->TempControlType(this->ControlZoneNum)) {
+            case DataHVACGlobals::SingleHeatingSetPoint: {
+                if (state.dataUnitarySystems->HeatingLoad && SensOutputOff > ZoneLoad &&
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:8316]: I think this logic can be simplified, maybe not part of this PR.

File: src/EnergyPlus/WindTurbine.cc [hunk 7]
```diff
@@ -688,46 +688,48 @@ namespace WindTurbine {
...
-                                        state.dataWindTurbine->WindTurbineSys(WindTurbineNum).Name);
-                    SetupOutputVariable(state,
-                                        "Generator Turbine Attack Angle",
-                                        OutputProcessor::Unit::deg,
-                                        state.dataWindTurbine->WindTurbineSys(WindTurbineNum).AngOfAttack,
-                                        OutputProcessor::SOVTimeStepType::System,
-                                        OutputProcessor::SOVStoreType::Average,
-                                        state.dataWindTurbine->WindTurbineSys(WindTurbineNum).Name);
-                }
+            switch (state.dataWindTurbine->WindTurbineSys(WindTurbineNum).rotorType) {
```
Comment (amirroth) [inline src/EnergyPlus/WindTurbine.cc:691]: Can start deploying the following pattern here. 

```cpp
auto &windTurbineSys = state.dataWindTurbineSys->WindTurbineSys(WindTurbineNum);
swtich (windTurbineSys.rotorType) {
...
```

Something to start getting used to.

File: src/EnergyPlus/WindTurbine.cc [hunk 8]
```diff
@@ -956,116 +958,114 @@ namespace WindTurbine {
...
-                        if (PowerCoeff > MaxPowerCoeff) {
-                            PowerCoeff = MaxPowerCoeff;
-                        }
-                        WTPower = 0.5 * LocalAirDensity * PowerCoeff * SweptArea * LocalWindSpeed_3;
-                    } else { // Simple approximation
-                        WTPower = 0.5 * LocalAirDensity * SweptArea * LocalWindSpeed_3 * MaxPowerCoeff;
+            switch (state.dataWindTurbine->WindTurbineSys(WindTurbineNum).rotorType) {
+            case RotorType::HAWT: { // Horizontal axis wind turbine
+                MaxPowerCoeff = state.dataWindTurbine->WindTurbineSys(WindTurbineNum).MaxPowerCoeff;
+                // Check if empirical constants are available
```
Comment (amirroth) [inline src/EnergyPlus/WindTurbine.cc:964]: `C1 = windTurbineSys.PowerCoeffC1;` etc.

File: src/EnergyPlus/WindowManager.cc [hunk 9]
```diff
@@ -7544,223 +7544,225 @@ namespace WindowManager {
...
-                                    GapVentType = "VentedOutdoor";
-                                }
-                                static constexpr std::string_view Format_713(" WindowMaterial:Gap:EquivalentLayer,{},{},{:.3R},{}\n");
-                                print(state.files.eio,
-                                      Format_713,
-                                      state.dataMaterial->Material(Layer).Name,
-                                      GasTypeName(state.dataMaterial->Material(Layer).GasType(1)),
-                                      state.dataMaterial->Material(Layer).Thickness,
-                                      GapVentType);
+                        switch (state.dataMaterial->Material(Layer).Group) {
```
Comment (amirroth) [inline src/EnergyPlus/WindowManager.cc:7547]: Same pattern

```cpp
auto &Material = state.dataMaterial->Material(Layer);
switch(material.Group) {
...
```

File: src/EnergyPlus/WindowManager.cc [hunk 10]
```diff
@@ -7544,223 +7544,225 @@ namespace WindowManager {
...
+                        case DataHeatBalance::MaterialGroup::WindowGas: {
+                            static constexpr std::string_view Format_702(" WindowMaterial:Gas,{},{},{:.3R}\n");
+                            print(state.files.eio,
+                                  Format_702,
```
Comment (amirroth) [inline src/EnergyPlus/WindowManager.cc:7551]: `material.Name, material.GasType(1), material.Thickness);`

File: src/EnergyPlus/WindowManager.cc [hunk 11]
```diff
@@ -7544,223 +7544,225 @@ namespace WindowManager {
...
+                                  state.dataMaterial->Material(Layer).Name,
+                                  state.dataMaterial->Material(Layer).Thickness,
+                                  state.dataMaterial->Material(Layer).Conductivity,
+                                  state.dataMaterial->Material(Layer).AbsorpThermal,
+                                  state.dataMaterial->Material(Layer).Trans,
+                                  state.dataMaterial->Material(Layer).TransVis,
+                                  state.dataMaterial->Material(Layer).ReflectShade);
+                        } break;
+                        case DataHeatBalance::MaterialGroup::WindowBlind: {
+                            BlNum = state.dataMaterial->Material(Layer).BlindDataPtr;
```
Comment (amirroth) [inline src/EnergyPlus/WindowManager.cc:7570]: Same pattern.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 12]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
 
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolDesDay = state.dataEnvrn->EnvironmentName;
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).HeatDesDay = state.dataEnvrn->EnvironmentName;
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DesHeatDens = state.dataEnvrn->StdRhoAir;
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DesCoolDens = state.dataEnvrn->StdRhoAir;
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).HeatDDNum = state.dataSize->CurOverallSimDay;
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolDDNum = state.dataSize->CurOverallSimDay;
-            }
+            state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolDesDay = state.dataEnvrn->EnvironmentName;
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Same pattern.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 13]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+            if (state.dataHeatBalFanSys->ZoneThermostatSetPointHi(CtrlZoneNum) > 0.0 &&
+                state.dataHeatBalFanSys->ZoneThermostatSetPointHi(CtrlZoneNum) > state.dataSize->ZoneSizThermSetPtHi(CtrlZoneNum)) {
+                state.dataSize->ZoneSizThermSetPtHi(CtrlZoneNum) = state.dataHeatBalFanSys->ZoneThermostatSetPointHi(CtrlZoneNum);
+            }
+            if (state.dataHeatBalFanSys->ZoneThermostatSetPointLo(CtrlZoneNum) > 0.0 &&
+                state.dataHeatBalFanSys->ZoneThermostatSetPointLo(CtrlZoneNum) < state.dataSize->ZoneSizThermSetPtLo(CtrlZoneNum)) {
+                state.dataSize->ZoneSizThermSetPtLo(CtrlZoneNum) = state.dataHeatBalFanSys->ZoneThermostatSetPointLo(CtrlZoneNum);
+            }
+            state.dataSize->ZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DesHeatSetPtSeq(TimeStepInDay) =
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Same pattern. Would really make this code easier to read.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 14]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
 
-    {
-        auto const SELECT_CASE_var(CallIndicator);
-
-        if (SELECT_CASE_var == DataGlobalConstants::CallIndicator::BeginDay) {
-
-            for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
+    switch (CallIndicator) {
+    case DataGlobalConstants::CallIndicator::BeginDay: {
+        for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Reduce scope of loop induction variables to loop itself: `for (int CtrlZoneNum ...`.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 15]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DOASSupTemp * FracTimeStepZone;
+            state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DOASSupHumRatSeq(TimeStepInDay) +=
+                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DOASSupHumRat * FracTimeStepZone;
+            state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DOASTotCoolLoadSeq(TimeStepInDay) +=
+                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DOASTotCoolLoad * FracTimeStepZone;
+        }
+    } break;
+    case DataGlobalConstants::CallIndicator::EndDay: {
+        // average some of the zone sequences to reduce peakiness
+        for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Is there a reason we are doing multiple loops, one for each type of sizing, rather than one loop with all the different sizing per zone?

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 16]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
-                state.dataZoneEquipmentManager->AvgData = 0.0;
-                MovingAvg(state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolZoneRetTempSeq,
-                          state.dataZoneEquipmentManager->NumOfTimeStepInDay,
-                          state.dataSize->NumTimeStepsInAvg,
-                          state.dataZoneEquipmentManager->AvgData);
-                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).CoolZoneRetTempSeq =
-                    state.dataZoneEquipmentManager->AvgData;
+            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+            // save the sequence values at the heating peak
+            for (TimeStepIndex = 1; TimeStepIndex <= state.dataZoneEquipmentManager->NumOfTimeStepInDay; ++TimeStepIndex) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Same comments as before.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 17]
```diff
@@ -1593,181 +1593,434 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+                TimeStepAtPeak = state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).TimeStepNumAtHeatMax;
+                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DesHeatCoilInTemp =
+                    OAFrac * state.dataSize->DesDayWeath(state.dataSize->CurOverallSimDay).Temp(TimeStepAtPeak) +
+                    (1.0 - OAFrac) * state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).ZoneTempAtHeatPeak;
+                state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).DesHeatCoilInHumRat =
+                    OAFrac * state.dataSize->DesDayWeath(state.dataSize->CurOverallSimDay).HumRat(TimeStepAtPeak) +
+                    (1.0 - OAFrac) * state.dataSize->CalcZoneSizing(state.dataSize->CurOverallSimDay, CtrlZoneNum).ZoneHumRatAtHeatPeak;
+            }
+            // save the sequence values at the cooling peak
+            for (TimeStepIndex = 1; TimeStepIndex <= state.dataZoneEquipmentManager->NumOfTimeStepInDay; ++TimeStepIndex) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Sorry I am piggy-backing on this PR with some other things, but you seem to be in the zone in terms of SELECT_CASE_var.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 18]
```diff
@@ -1972,1022 +2130,835 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+                      state.dataSize->CalcFinalZoneSizing(I).DesCoolVolFlow);
+            }
+            print(state.files.zsz, "\n");
+            state.files.zsz.close();
+        }
 
+        // Move data from Calc arrays to user modified arrays
+
+        for (std::size_t i = 0; i < state.dataSize->ZoneSizing.size(); ++i) {
+            auto &z(state.dataSize->ZoneSizing[i]);
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Yes, this is the pattern I am trying to propagate elsewhere.

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 19]
```diff
@@ -1972,1022 +2130,835 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
+                      state.dataSize->SizingFileColSep,
+                      state.dataSize->CalcFinalZoneSizing(I).DesCoolVolFlow);
+            }
+            print(state.files.zsz, "\n");
+            state.files.zsz.close();
+        }
 
+        // Move data from Calc arrays to user modified arrays
+
+        for (std::size_t i = 0; i < state.dataSize->ZoneSizing.size(); ++i) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: would `state.dataSize->ZoneSizing[i] = state.dataSize->CalcZoneSizing[i]` not work?

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 20]
```diff
@@ -1972,1022 +2130,835 @@ void UpdateZoneSizing(EnergyPlusData &state, DataGlobalConstants::CallIndicator
...
-                    state.dataSize->FinalZoneSizing(CtrlZoneNum).DesCoolVolFlow > 0.0) {
-                    TotCoolSizMult = (state.dataSize->FinalZoneSizing(CtrlZoneNum).InpDesCoolAirFlow /
-                                      state.dataSize->FinalZoneSizing(CtrlZoneNum).DesCoolVolFlow) *
-                                     state.dataSize->FinalZoneSizing(CtrlZoneNum).CoolSizingFactor;
-                    // If no user specified cooling design air flow rate input, use the user specified szing factor
+        }
+
+        for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
+            if (!state.dataZoneEquip->ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+            for (TimeStepIndex = 1; TimeStepIndex <= state.dataZoneEquipmentManager->NumOfTimeStepInDay; ++TimeStepIndex) {
```
Comment (amirroth) [inline src/EnergyPlus/ZoneEquipmentManager.cc]: Is all this copying necessary?

File: src/EnergyPlus/SteamBaseboardRadiator.cc (hunk 2 above)
Comment (Myoldmopar) [reply src/EnergyPlus/SteamBaseboardRadiator.cc:199]: This didn't get converted back to a normal if/else...perhaps a follow-up pass could find any straggling single-path switch blocks and convert them to plain IFs.  This PR looks like it is ready so I won't hold it up just for this...but make a note of it @dareumnam

File: src/EnergyPlus/TARCOGOutput.cc [hunk 21]
```diff
@@ -349,25 +349,29 @@ void WriteInputArguments(EnergyPlusData &state,
...
-                print(InArgumentsFile, Format_10804, i, LayerType(i));
-            } else if (SELECT_CASE_var == TARCOGLayerType::VENETBLIND_VERT) { // Vertical venetian blind
-                print(InArgumentsFile, Format_10810, i, LayerType(i));
-            } else if (SELECT_CASE_var == TARCOGLayerType::SPECULAR) { // Specular layer
-                if (nslice(i) <= 1) {
-                    print(InArgumentsFile, Format_10802, i, LayerType(i)); // Monolithic glass
-                } else {
-                    print(InArgumentsFile, Format_10803, i, LayerType(i)); // Laminated layer
-                }
+        switch (LayerType(i)) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/TARCOGOutput.cc:352]: So much better!

File: src/EnergyPlus/ThermalComfort.cc (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/ThermalComfort.cc:2462]: @dareumnam take note of these spots as well and if appropriate address them in your next PR.

File: src/EnergyPlus/WaterUse.cc [hunk 22]
```diff
@@ -1496,42 +1496,45 @@ namespace WaterUse {
...
-            {
-                auto const SELECT_CASE_var(this->HeatRecoveryConfig);
-                if (SELECT_CASE_var == HeatRecoveryConfigEnum::Plant) {
-                    this->RecoveryMassFlowRate = this->HotMassFlowRate;
-                } else if (SELECT_CASE_var == HeatRecoveryConfigEnum::Equipment) {
-                    this->RecoveryMassFlowRate = this->ColdMassFlowRate;
-                } else if (SELECT_CASE_var == HeatRecoveryConfigEnum::PlantAndEquip) {
-                    this->RecoveryMassFlowRate = this->TotalMassFlowRate;
-                }
+            switch (this->HeatRecoveryConfig) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/WaterUse.cc:1499]: I don't know what else to say as I review this, it's just so much better.  Every SELECT_CASE_var removed brings one extra smile to someone, somewhere.

**Review by Myoldmopar — APPROVED** [review]
Yep, all good here.

**Comment by Myoldmopar** [comment]
Tested with develop pulled in and it's working great.  No issues.  Merging, thanks @dareumnam for the changes and @amirroth for all the reviews!

**Comment by dareumnam** [comment]
@Myoldmopar @amirroth Thanks for reviewing. I'm addressing comments on my local branch and I'll create a follow-up PR asap.

**Comment by Myoldmopar** [comment]
By midnight, please.

**Comment by Myoldmopar** [comment]
:laughing: :laughing: Good night @dareumnam !

**Comment by dareumnam** [comment]
Haha Have a good night! @Myoldmopar
